torch/_export/passes/replace_set_grad_with_hop_pass.py

```diff
@@ -132,22 +132,34 @@ def replace_set_grad_with_hop_pass(gm: torch.fx.GraphModule):
         if _is_set_grad_enabled_node(node):
             need_replacing = True
 
-    if not need_replacing:
-        return gm
+    if need_replacing:
+        new_gm = sequential_split(gm, _is_set_grad_enabled_node)
 
-    new_gm = sequential_split(gm, _is_set_grad_enabled_node)
-
-    def _maybe_inline_or_replace_with_hop(node: torch.fx.Node):
-        if _is_set_grad_enabled_sub_mod(node, omit_if_same_with_ambient=True):
-            _replace_with_hop(node)
-        else:
-            _remove_set_grad_and_inline(node)
+        def _maybe_inline_or_replace_with_hop(node: torch.fx.Node):
+            if _is_set_grad_enabled_sub_mod(node, omit_if_same_with_ambient=True):
+                _replace_with_hop(node)
+            else:
+                _remove_set_grad_and_inline(node)
+
+        nodes_map(
+            list(new_gm.graph.nodes),
+            lambda node: (
+                _maybe_inline_or_replace_with_hop(node)
+                if node.op == "call_module"
+                else node
+            ),
+        )
+    else:
```

Comment:
NIT: Could you make it a utility function? so that we don't have to do if/else branch. Something like:
```
def get_new_graph(gm):
       if need_replacing: return bar(gm)
       return gm
       
```

Reply:
Thanks for the review! will do